File: hoiiday-backend/src/main/java/com/example/hoiiday/service/RulesService.java
package com.example.hoiiday.service;

import com.example.hoiiday.DTO.RulesDTO;

import java.util.List;

public interface RulesService {
    List<RulesDTO> getAllRules();
    RulesDTO getRuleById(Long id);
    RulesDTO createRule(RulesDTO rulesDTO);
    RulesDTO updateRule(Long id, RulesDTO rulesDTO);
    void deleteRule(Long id);
}

File: hoiiday-front/hoiiday-frontend/src/app/services/rules.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Rules } from '../model/rules.model';

@Injectable({
  providedIn: 'root',
})
export class RulesService {
  private apiUrl = '/api/rules'; // Adjust if your backend URL is different

  constructor(private http: HttpClient) {}

  getAllRules(): Observable<Rules[]> {
    return this.http.get<Rules[]>(this.apiUrl);
  }

  getRuleById(id: number): Observable<Rules> {
    return this.http.get<Rules>(`${this.apiUrl}/${id}`);
  }

  createRule(rule: Rules): Observable<Rules> {
    return this.http.post<Rules>(this.apiUrl, rule);
  }

  updateRule(id: number, rule: Rules): Observable<Rules> {
    return this.http.put<Rules>(`${this.apiUrl}/${id}`, rule);
  }

  deleteRule(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}